### PR TITLE
add Learneth plugin

### PR DESCRIPTION
**Pull request by drafish**
# Remix LearnEth Plugin

## Available Scripts

In the project directory, you can run:

### `npm run serve:plugin --plugin=learneth`

Runs the app in the development mode.\
Open [http://localhost:2024](http://localhost:2024) to view it in the browser.

The page will reload if you make edits.\
You will also see any lint errors in the console.

### `npm run build:plugin --plugin=learneth`

Builds the app for production to the `dist/apps/learneth` folder.

You can test it on chinese mirror site of remix ide -- https://remix.learnblockchain.cn/?#activate=LearnEth

Here is the url of learneth on mirror site -- https://remix.learnblockchain.cn/plugins/learneth/index.html

### CODE REVIEW

**Review by ryestew — COMMENTED** [review]
From my point of view, the plugin is working.

File: apps/learneth/src/assets/logo-background.svg
Comment (LianaHus) [inline apps/learneth/src/assets/logo-background.svg:1]: we have it in assets folder.

**Review by yann300 — APPROVED** [review]

**Review by LianaHus — CHANGES_REQUESTED** [review]
Please, whatever is possible, do not use custom CSS classes but bootstrap once.

File: apps/learneth/src/assets/logo-background.svg
Comment (drafish) [reply apps/learneth/src/assets/logo-background.svg:1]: I tried to import the logo like this 
```ts
import logo from '../../../../remix-ide/src/assets/img/remixLogo.webp'
```

Then I got this error when I was building
image

I tried this way which works, check below
```html
<img src="https://remix.ethereum.org/assets/img/remixLogo.webp" />
```

But it's not elegant. Will that be ok? @LianaHus

**Comment by drafish** [comment]
> Please, whatever is possible, do not use custom CSS classes but bootstrap once.

What you mean by `bootstrap once`? Are you suggesting that use bootstrap classes only, and not use custom classes?

BTW, I got the custom CSS classes from the old learneth plugin.  @LianaHus

File: apps/learneth/src/pages/Home/index.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                        handleClick(item.id);
+                      }}
+                    >
+                      {selectedRepo.entities[item.id].name}
+                    </a>
+                    <Link
+                      to={`/list?id=${item.id}`}
+                      className="text-decoration-none float-right"
+                    >
+                      <FontAwesomeIcon icon={faPlayCircle} size="lg" />
```
Comment (LianaHus) [inline apps/learneth/src/pages/Home/index.tsx:83]: please use the same coding style as we do in other places.
example https://github.com/ethereum/remix-project/blob/master/libs/remix-ui/run-tab/src/lib/components/contractGUI.tsx#L366

File: apps/learneth/src/pages/StepList/index.scss [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  text-align: left;
+  font-size: 1.2rem !important;
+  word-break: break-word;
+}
+section {
+  flex: 1;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+
```
Comment (LianaHus) [inline apps/learneth/src/pages/StepList/index.scss:35]: please remove empty spaces

File: apps/learneth/src/assets/logo-background.svg
Comment (LianaHus) [reply apps/learneth/src/assets/logo-background.svg:1]: webp is also fine. depending on where you use it. does it work for both types of themes?

**Comment by LianaHus** [comment]
Please, whatever is possible, do not use custom CSS classes but bootstrap once.

> > Please, whatever is possible, do not use custom CSS classes but bootstrap once.
> 
> What you mean by `bootstrap once`? Are you suggesting that use bootstrap classes only, and not use custom classes?
> 
> BTW, I got the custom CSS classes from the old learneth plugin. @LianaHus

yes. we are removing custom classes everywhere and using bootstrap classes

File: apps/learneth/src/pages/StepList/index.scss (hunk 2 above)
Comment (drafish) [reply apps/learneth/src/pages/StepList/index.scss]: removed here, and other files too.

File: apps/learneth/src/pages/Home/index.tsx (hunk 1 above)
Comment (drafish) [reply apps/learneth/src/pages/Home/index.tsx:67]: I resaved all files. Now it's using remix's coding style.

**Comment by drafish** [comment]
> Please, whatever is possible, do not use custom CSS classes but bootstrap once.
> 
> > > Please, whatever is possible, do not use custom CSS classes but bootstrap once.
> > 
> > 
> > What you mean by `bootstrap once`? Are you suggesting that use bootstrap classes only, and not use custom classes?
> > BTW, I got the custom CSS classes from the old learneth plugin. @LianaHus
> 
> yes. we are removing custom classes everywhere and using bootstrap classes

Can I do this in next PR? There are a lot custom css classes to be removed. That may take a while. @LianaHus

File: apps/learneth/src/assets/logo-background.svg
Comment (drafish) [reply apps/learneth/src/assets/logo-background.svg:1]: I deleted the assets folder in learneth, get the assets from remix-ide folder.

File: apps/learneth/src/pages/Home/index.tsx (hunk 1 above)
Comment (LianaHus) [reply apps/learneth/src/pages/Home/index.tsx:65]: I'm sorry I still see FontAwesomeIcon  in the code.

Comment (drafish) [reply apps/learneth/src/pages/Home/index.tsx]: > I'm sorry I still see FontAwesomeIcon in the code.

just removed. please check again. thanks @LianaHus

**Review by LianaHus — APPROVED** [review]
Approving the PR. Please submit a new one to remove custom CSS and use bootstrap classes instead.

**Pull request merged**